let Joi,CustomErrorHandler;_ae6‍.x([["default",()=>_ae6‍.o]]);_ae6‍.w("joi",[["default",["Joi"],function(v){Joi=v}]]);_ae6‍.w("../../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const registerController = {
  async register(req, res, next) {
    // Validation Schema
    const registerSchema = Joi.object({
      name: Joi.string().min(3).max(30).required(),
      email: Joi.string().email().required(),
      password: Joi.string()
        .pattern(new RegExp("^[a-zA-Z0-9]{3,30}$"))
        .required(),
      repeat_password: Joi.ref("password"),
    });
    // End of Validation Schema

    const {error} = registerSchema.validate(req.body);
    if (error) {
      return next(error);
    }

    // Check if user is exists

    try {
      const exist = await User.exists({email: req.body.email});
      if (exist) {
        return next(CustomErrorHandler.alreadyExist("This email is already taken."));
      }
    } catch (error) {}

    res.send({
      message: "OK",
      data: req.body,
    });
  },
};

_ae6‍.d(registerController);
